# Enforce Rules of Hooks

## Why?

> xx

## bad

```scss

```

## good

```scss

```

## 参考:

- [accessible-emoji](https://github.com/jsx-eslint/eslint-plugin-react/blob/c42b624d0fb9ad647583a775ab9751091eec066f/docs/rules/accessible-emoji)
