# boolean 值为 true 的属性只需要写属性名即可

## Why?

> 统一风格

## bad

```jsx
var Hello = <Hello personal={true} />;
```

## good

```jsx
var Hello = <Hello personal />;
```

## 参考

- [](https://stylelint.io/user-guide/rules/list/)
