# 必须在@规则前换行

## Why?

> 统一风格,提高可阅读性

## bad

```css
a {
}
@media {
}
```

## good

```css
a {
}

@media {
}
```

## 参考

- [at-rule-empty-line-before](https://stylelint.io/user-guide/rules/list/at-rule-empty-line-before)
