# 必须在注释符号里面添加一个空格

## Why?

> 统一风格

## bad

```scss
/*注释*/
```

## good

```scss
/* 注释 */
```

## 参考

- [comment-whitespace-inside](https://stylelint.io/user-guide/rules/list/comment-whitespace-inside)
