# 必须在注释前面换行

## Why?

> 统一风格

## bad

```scss
div {
}
/* 注释 */
```

## good

```scss
div {
}

/* 注释 */
```

## 参考

- [comment-empty-line-before](https://stylelint.io/user-guide/rules/list/comment-empty-line-before)
