# 禁止出现空的注释

## Why?

> 空的注释没有存在的意义,请删除或添加注释内容.

## bad

```css
/* */
```

## good

```css
/* balabalabala */
```

## 参考

- [comment-no-empty](https://stylelint.io/user-guide/rules/list/comment-no-empty)
