# 禁止出现空的样式文件

## Why?

> 空文件没有实际意义,请删除掉或者添加注释为什么保留空文件

## bad

```css
\t\n
```

## good

```css
/* balabalabala */
```

## 参考

- [no-empty-source](https://stylelint.io/user-guide/rules/list/no-empty-source)
