# 禁止出现多余的分号

## Why?

> 多余的分号没实际意义,一般是因为输入错误导致,请删掉

## bad

```css
a {
  color: pink;
}
```

## good

```css
a {
  color: pink;
}
```

## 参考

- [no-extra-semicolons](https://stylelint.io/user-guide/rules/list/no-extra-semicolons)
