# @规则必须使用小写字母

## Why?

> scss 中只能使用小写,在所有语言中统一规范,有利于提高可阅读性

## bad

```
@MEDIA screen {
}
```

## good

```scss
@media screen {
}
```

## 参考

- [at-rule-name-case](https://stylelint.io/user-guide/rules/list/at-rule-name-case)
