# 必须使用小写烤串命名 ID 选择器

## Why?

> CSS 是大小写不敏感语言,统一风格

## bad

```scss
#Foo-bar {
}
```

## good

```scss
#foo-bar {
}
```

## 参考

- [selector-id-pattern](https://stylelint.io/user-guide/rules/list/selector-id-pattern)
