# 必须使用小写瓢节点选择器

## Why?

> HTML,CSS 是大小写不敏感语言,统一风格

## bad

```scss
A {
}
```

## good

```scss
a {
}
```

## 参考

- [selector-type-case](https://stylelint.io/user-guide/rules/list/selector-type-case)
