# 禁止使用大写标注属性值

## Why?

> 统一风格

## bad

```scss
a {
  display: BLOCK;
}
```

## good

```scss
a {
  display: block;
}
```

## 参考

- [value-keyword-case](https://stylelint.io/user-guide/rules/list/value-keyword-case)
