# 组件名必须由多个单词构成

## Why?

> html 标签全是一个单词,为了一目了然的区分 vue 组件和原生 html 标签,组件统一使用多个单词.

## bad

```vue
<script>
export default {
  name: "Order",
};
</script>
```

## good

```vue
<script>
export default {
  name: "CustomerOrder",
};
</script>
```

## 参考

- [vue/multi-word-component-names](https://eslint.vuejs.org/rules/multi-word-component-names.html)
