# 禁止使用非法的伪类选择器

## Why?

> 语法错误,请修改成正确的

## bad

```css
a:hove {
}
```

## good

```css
a:hover {
}
```

## 参考

- [selector-pseudo-class-no-unknown](https://stylelint.io/user-guide/rules/list/selector-pseudo-class-no-unknown)
