# 禁止使用非法的标签选择器

## Why?

> 语法错误,自定义节点除外,请修改成正确的

## bad

```css
tag {
}
```

## good

```css
a {
}
```

## 参考

- [selector-type-no-unknown](https://stylelint.io/user-guide/rules/list/selector-type-no-unknown)
