# 必须导入已存在的命名模块

## Why?

> 导入的模块必须已经被导出,否则在运行时会抛出异常

outer.js

```js
export const name = "zmn";
```

## bad

```js
import { name1 } from "./outer.js";
```

## good

```js
import { name } from "./outer.js";
```

## 参考:

- [import/named](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md)
